src/util/records.py

```diff
@@ -1479,7 +1479,15 @@ def replace_record_references_batch(cr, id_mapping, model_src, model_dst=None, r
             continue
         if ir.company_dependent_comodel:
             if ir.company_dependent_comodel == model_src:
-                assert model_src == model_dst
+                if model_src != model_dst:
```

Comment:
Can't we just remove the `assert`? We can check instead that _all_ keys were replaced. In such cases we can "trust" that the model of the underlying field is being changed. I find it odd to update ir_model_fields relation column here...
@KangOl wdyt?

Reply:
we can remove the assertion and trust that the relation will change later, I only added this part to be consistent with the other IR where we change the res_model, and I share the same concern

Reply:
> I only added this part to be consistent with the other IR where we change the res_model

Can you put a link to it please?

Reply:
https://github.com/odoo/upgrade-util/blob/c47b904633f1e3133c277b339ffeb534bffa056d/src/util/records.py#L1508-L1531

Reply:
Well indeed. To be aligned we could assume that the "ir.res_model" of company dependent fields (jsonb version) is the model of the field. I think we could still add some sort of check after we change the model to ensure all values (ids) are present in the referenced table. Can you do a quick test to see in 18 what happens if the value points to an invalid key (id)? Is it ignored? Does it raises any error?

Reply:
Did a small test, if the id does not exist then an [error](https://github.com/odoo/odoo/blob/3e4c4516f7d238e76e2e9b7b71a807f7cc7d55b3/odoo/fields.py#L1267-L1271) is raised and the view is inaccessible, i'll add the check, and I will do the same test for regular IR fields